src/main/java/clzzz/helper/commons/util/DateTimeUtil.java

```diff
@@ -8,30 +10,56 @@
  */
 public class DateTimeUtil {
 
+    // constants
+
     // datetime-related
     public static final String DATETIME_PATTERN = "d/M/uuuu HHmm";
     //used to check the format of date time
     public static final DateTimeFormatter DATETIME_FORMAT =
-            DateTimeFormatter.ofPattern(DATETIME_PATTERN).withResolverStyle(ResolverStyle.LENIENT);
-    // used to check if the time is valid
-    public static final String TIME_PATTERN = "HHmm";
-    public static final DateTimeFormatter STRICT_TIME_FORMAT =
-            DateTimeFormatter.ofPattern(TIME_PATTERN).withResolverStyle(ResolverStyle.STRICT);
-
+            DateTimeFormatter.ofPattern(DATETIME_PATTERN).withResolverStyle(ResolverStyle.STRICT);
```

Comment:
The formatter with different resolverstyle is to differentiate invalid date and date with invalid format. Using LENIENT resolverstyle, the app will throw error when format is invalid. Using STRICT, it will throw error when either format is invalid or date is invalid. In order to throw different error message according to the user input, we use LENIENT to filter those with valid input format, and use STRICT to further narrow down to  those with valid date. We cannot achieve the purpose using STRICT only... Can u revert the change here..? And also those methods in ParserUtil/SlotParserUtil/DOB/DateTime that make uses of these formatters?

Reply:
hmmm but not sure if we need to differentiate between invalid format, and invalid date value. boundaries for those 2 are not very clear, so sometimes it's quite hard to tell whether a given string is invalid format, or invalid date value, or both, that's why i just combined them together into just "invalid date".

Reply:
regardless i think that any validating of dates and times should be deferred to a single place in `DateTimeUtil` instead of in `Slot`/`DateOfBirth`/etc. i'll probs fix that in a separate pr

Reply:
I'm not very sure where should we put the methods. I put it in Slot/DOB cuz other classes do that, like Name, Species, etc.

Reply:
Ok then if we want to group invalid date and invalid format together, I think we need to change the error message to include both possible errors. Now I think the message constraints only reflect there is an error in format.

Reply:
how about

    DateOfBirth.MESSAGE_CONSTRAINTS = 
            "Date of birth must be valid, and follow the format of " + DateTimeUtil.DATE_PATTERN + ".";

and

    DateTime.MESSAGE_CONSTRAINTS = 
            "Date and time must be valid, and follow the format of " + DateTimeUtil.DATETIME_PATTERN + ".";
?

Reply:
I think ok!:D but maybe just type out the format? cuz user may not know what "uuuu" mean.

Reply:
Sigh, Java problems